src/main/java/com/openmrs/migrator/core/utilities/FileIOUtilities.java

```diff
@@ -156,4 +175,148 @@ public boolean removeDirectory(File directoryToBeDeleted) throws InvalidParamete
+              }
+              if (configLine.contains("password=")) {
+                password = configLine.split("=")[1];
+              }
+              if (configLine.contains("host=")) {
+                host = configLine.split("=")[1];
+              }
+              if (configLine.contains("port=")) {
+                port = configLine.split("=")[1];
+              }
+            });
+
+    new PrintWriter(kettlePath.toFile()).close();
+
+    writeToFile(
+        kettlePath.toFile(),
+        "ETL_SOURCE_DATABASE=" + dataBaseName,
+        "ETL_DATABASE_HOST=" + host,
+        "ETL_DATABASE_PORT=" + port,
+        "ETL_DATABASE_USER=" + username,
+        "ETL_DATABASE_PASSWORD=" + password);
+  }
+
+  private File getKettlePropertiesLocation() throws IOException {
+    logger.info("Getting the default location of kettle.properties");
+    String homeDirectory = System.getProperty("user.home");
+
+    Path kettleDir = Paths.get(homeDirectory + "/" + KETTLE_DIR);
+    Path kettleFile = Paths.get(kettleDir + "/" + KETTLE_PROPERTIES);
+
+    if (!Files.exists(kettleDir)) {
+      Files.createDirectories(kettleDir);
+      Files.createFile(kettleFile);
+    }
+    if (!Files.exists(kettleFile)) {
+      Files.createFile(kettleFile);
+    }
+
+    File file = new File(homeDirectory + "/" + KETTLE_DIR + "/" + KETTLE_PROPERTIES);
+    return file;
+  }
+
+  public List<String> getAllDataBaseNamesFromConfigFile()
+      throws FileNotFoundException, IOException {
+    logger.info("retrieving all data base  names in " + settingProperties + " file");
+    List<String> names = new ArrayList<>();
+    try (BufferedReader br = new BufferedReader(new FileReader(settingProperties.toFile()))) {
+      String line;
+      while ((line = br.readLine()) != null) {
+        if (line.contains("database_name")) {
+          names.add(line.split("=")[1]);
+        }
+      }
+    }
+    return names;
+  }
+
+  private void writeToFile(File file, String... contents) throws IOException {
+    try (BufferedWriter bw = new BufferedWriter(new FileWriter(file))) {
+      for (String content : contents) {
+        bw.append(content);
+        bw.newLine();
+      }
+      bw.flush();
+    }
+  }
+
+  public void fillConfigFile() throws IOException {
+
+    Map<String, String> connDB = ConsoleUtils.readSourceDBConn(System.console());
+
+    writeToFile(
+        settingProperties.toFile(),
+        "username=" + connDB.get("username="),
+        "password=" + connDB.get("password="),
+        "host=" + connDB.get("host="),
+        "port=" + connDB.get("port="));
+  }
+
+  public void fillConfigFile(String user, String password, String host, String port)
+      throws IOException {
+
+    writeToFile(
+        settingProperties.toFile(),
+        "username=" + user,
+        "password=" + password,
+        "host=" + host,
+        "port=" + port);
+  }
+
+  public List<Path> listFiles(Path path) throws IOException {
```

Comment:
this belongs here, but pls add unit tests

Reply:
done